sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -270,22 +250,20 @@ namespace Azure { namespace Core { namespace Http {
      * @brief Set URL path.
      *
      * @param path URL path.
-     * @param isPathEncoded `true` if \p fragment is URL-encoded, `false` otherwise.
      */
-    void SetPath(const std::string& path, bool isPathEncoded = false)
-    {
-      m_encodedPath = isPathEncoded ? path : EncodePath(path);
-    }
+    void SetPath(const std::string& encodedPath) { m_encodedPath = encodedPath; }
 
     /**
-     * @brief Set URL fragment.
+     * @brief Set the Query Parameters object from an existing query parameter map.
      *
-     * @param fragment URL fragment.
-     * @param isFragmentEncoded `true` if \p fragment is URL-encoded, `false` otherwise.
+     * @remark Keys and values in \p queryParameters are expected to be url-encoded.
+     *
+     * @param queryParameters
      */
-    void SetFragment(const std::string& fragment, bool isFragmentEncoded = false)
+    void SetQueryParameters(std::map<std::string, std::string> queryParameters)
```

Comment:
Do we really need this method?

Reply:
We need it to restore query parameters after each try on the retry policy.  So before the next try we have the original query parameters set again